tests/core/evaluation/test_marker_stats.py

```diff
@@ -135,8 +135,10 @@ def test_process_results_per_session(seed: int):
                 preceding_user_turn_numbers_used_per_marker[marker][idx]
             )
             for stat_name, stat_value in expected_stats.items():
-                assert pytest.approx(
-                    stats.session_results[marker][stat_name][idx], stat_value
+                assert np.isclose(
```

Comment:
@ka-bu I requested your review to check that this change in this unit test is valid. The current update to `pytest` breaks the test in its old unit test (see[ CI workflow](https://github.com/RasaHQ/rasa/runs/5993986933?check_suite_focus=true))

Reply:
Looking at the error, it looks like the original test was not correctly written:

```
>                   assert pytest.approx(
                        stats.session_results[marker][stat_name][idx], stat_value
                    )
E                   AssertionError: approx() is not supported in a boolean context.
E                   Did you mean: `assert a == approx(b)`?
```

as in 

```python
assert pytest.approx(stats.session_results[marker][stat_name][idx], stat_value)
```
just asserts that the result of `pytest.approx()` is a non `None` object, not whether the expected stat value is equal to the actual stat value.

Why not just rewrite this as 

```python
assert pytest.approx(stats.session_results[marker][stat_name][idx]) == stat_value
```

Or is it the case that we are expecting NaNs and need the NaNs equality check in `np.isclose`?

Reply:
Oh yes, so the `pytest.approx` assetion fails unless I add the `nan_ok=True` kw arg, which defaults to `False` otherwise. I'll change to this now.

Reply:
yep, that should've been an np.isclose 👍  And yes, nans can appear because If there are no signals collected, all stats (except the count which will be 0) over those non-existing signals etc will be nan

Reply:
Thanks for the follow-up @ka-bu 👍🏼 
Should I revert to using `np.isclose` or the current usage of `pytest.approx` is also alright?

Reply:
Guess the intention had been to use `np.isclose(a,b)` to represent "a aproximately b" but then pytest.approx was called instead of np.isclose, and that is a problem because the notation is different with pytest.approx, i.e. "pytest.approx(a) = b" (There is a comparison of np.iclose and pytest.approx at the end of the docs [here](https://docs.pytest.org/en/6.2.x/reference.html#pytest-approx)). 
Fixing the notation and adding the nan_ok=True as you suggested is ok 👍

Reply:
Great 🙌🏼  let me know if this can be approved 😄